src/evergreen/evg_config_utils.rs

```diff
@@ -517,6 +534,61 @@ impl EvgConfigUtils for EvgConfigUtilsImpl {
             .collect()
     }
 
+    /// Determine burn in tag build variants for the given build variant.
+    ///
+    /// # Arguments
+    ///
+    /// * `build_variant` - Build Variant to query.
+    /// * `build_variant_map` - A map of build variant names to their definitions.
+    ///
+    /// # Returns
+    ///
+    /// List of build variant names to use for burn in tags.
+    fn resolve_burn_in_tag_build_variants(
+        &self,
+        build_variant: &BuildVariant,
+        build_variant_map: &HashMap<String, &BuildVariant>,
+    ) -> Vec<String> {
+        let mut burn_in_build_variants = self
+            .lookup_and_split_by_whitespace_build_variant_expansion(
+                BURN_IN_TAG_INCLUDE_BUILD_VARIANTS,
+                build_variant,
+            );
+        if self
+            .lookup_build_variant_expansion(
+                BURN_IN_TAG_REQUIRED_AND_SUGGESTED_BUILD_VARIANTS,
+                build_variant,
+            )
+            .unwrap_or_default()
+            == "true"
+        {
+            burn_in_build_variants.extend(
+                build_variant_map
+                    .iter()
+                    .filter_map(|(name, build_variant)| {
+                        let display_name = build_variant.display_name.as_ref().unwrap();
+                        if display_name.starts_with('!') || display_name.starts_with('*') {
```

Comment:
We could add it here in `evg_config_utils.rs` but the way things are currently designed, it could only be used here. We wouldn't easily be able to reuse it in `evg_config.rs` at the [other place](https://github.com/trahman1318/mongo-task-generator/blob/abc85f6f108d451330ff05d8fb7319f2a84a8aad/src/evergreen/evg_config.rs#L63). That brings up some questions about should a `EvgProjectConfig` struct from `evg_project.rs` "have a" `EvgConfigUtils` from `evg_config_utils.rs`, etc.

It seems to me like these are intentionally decoupled but I could be overthinking it. Maybe the real solution would be to put some of this logic into [shrub-rs](https://github.com/dbradf/shrub-rs/blob/b2944d31269d2fc69bca0b4acc2398696f998fb6/src/models/variant.rs#LL13C6-L13C6)? Anyways, I'd like to leave this for now & if we need to I think we can pull this out in the future.

Reply:
Moving this logic to `shrub-rs` seems reasonable